flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/streaming/connectors/elasticsearch/ElasticsearchSinkBase.java

```diff
@@ -234,4 +255,36 @@ private void checkErrorAndRethrow() {
 			throw new RuntimeException("An error occured in ElasticsearchSink.", cause);
 		}
 	}
+
+	private void retry(T value) throws Exception {
+		int retryCounter = 1;
+
+		while (retryCounter <= connectionRetries) {
+			if (bulkProcessor != null) {
+				bulkProcessor.close();
+				bulkProcessor = null;
+			}
+
+			if (client != null) {
+				client.close();
+			}
+
+			try {
+				open(null);
+				elasticsearchSinkFunction.process(value, getRuntimeContext(), requestIndexer);
+			} catch (Exception ex) {
+				if (client instanceof TransportClient && !callBridge.isConnected(((TransportClient) client))) {
+					TimeUnit.SECONDS.sleep(3);
```

Comment:
Should this be configurable?
Also, could you explain a bit on why you've chosen 3 seconds?